Improve the flat terms sidebar panel's user information

imath opened the pull request:
It's not obvious for a user a comma needs to be used to separate tags.
Using the `separate_items_with_commas` label to display it under the tags input should help him to understand how he can do it.

## Description
I've added a paragraph under the flat terms input field displaying the `separate_items_with_commas` label of the taxonomy object.

## How has this been tested?
Tested using VVV, WordPress 4.9.8, Gutenberg master.

## Screenshots
![separate-tags](https://user-images.githubusercontent.com/1834524/49557089-c977ed00-f906-11e8-8e0d-83e2af987838.png)

The "how to" paragraph.

## Types of changes
Bug fix (for new users it can be challenging)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

talldan reviewed (changes requested):
Hi @imath - thanks for another contribution!

I think the implementation of this should be moved into FormTokenField where there's already something similar, and there are a couple of minor styling bits that need to be sorted out.

imath committed:
Improve the flat terms sidebar panel's user help
It's not obvious for a user a comma needs to be used to separate tags. Using the 'separate_items_with_commas' Post type object label to display some help under the tags input should let him figure it out.
I've also applied @talldan's recommandation adding a new help property to the FormTokenField component to transport the text for the help.

imath commented on packages/editor/src/components/post-taxonomies/flat-term-selector.js:
```diff
@@ -185,23 +185,33 @@ class FlatTermSelector extends Component {
+		const howTo = get(
+			taxonomy,
+			[ 'labels', 'separate_items_with_commas' ],
+			_x( 'Separate terms with commas', 'how to separate term items' )
```
Hi @talldan thanks a lot for your review and advices. I've just applyed them and updated the pull request.

talldan reviewed (commented):
Thanks for making those changes so quickly! That looks good. There are a couple of other documentation things I've mentioned in the third comment that I think also should be addressed, but in terms of the code I'm happy 😄 

I see the 'needs design feedback' label was also added, hopefully a designer can have a look at some point.

afercia commented:
> Removing the classname makes it visible. I'm not sure why that decision was made over having something visible, it might be nice to find out.

Because designers didn't want it visible 🙂Personally, I'm all for making it visible. At this point I'd make also the default help visible and I guess code could be simplified to just use something like:

`help || __( 'Separate with commas' )`

gziolo commented:
@imath do you plan to revisit this PR? It wasn't updated for almost 2 months. I'm doing triage of all open PRs and therefore will mark it as `Stale` to indicate that it doesn't need to be reviewed until it's refreshed.

imath commented:
@gziolo Thanks a lot for the ping. I'll revisit the PR but not so soon. I'll ping you or @talldan when it will be refreshed. (sorry for not being too much available at the moment).

gziolo commented:
> I'll ping you or @talldan when it will be refreshed. (sorry for not being too much available at the moment).

That's totally fine, thanks for letting us know and your contribution so far 👍

mapk commented:
@imath I'm closing this PR for now because it looks like there's been over 2 months of no progress. If you choose to open it back up again, we can.
